pkg/controller/servicemesh/controlplane/htpasswd.go

```diff
@@ -111,13 +136,14 @@ func (r *controlPlaneInstanceReconciler) patchGrafanaConfig(ctx context.Context,
 
 	log.Info("patching Grafana-Prometheus link", object.GetKind(), object.GetName())
 
-	rawPassword, err := r.getRawHtPasswd(ctx)
+	authToken, err := r.getHtauthToken(ctx)
 	if err != nil {
 		return err
 	}
+	b64authToken := base64.StdEncoding.EncodeToString([]byte(authToken))
 
-	re := regexp.MustCompile("(?s)(basicAuthPassword:).*?\n")
-	dsYaml = re.ReplaceAllString(dsYaml, fmt.Sprintf("${1} %s\n", rawPassword))
+	re := regexp.MustCompile("(?s)({AuthToken}).*?\n")
+	dsYaml = re.ReplaceAllString(dsYaml, fmt.Sprintf("Basic %s'\n", b64authToken))
```

Comment:
Could we utilize `unstructured` API to set `secureJsonData.httpHeaderValue1` instead of this regex?